desc/io/input_reader.py

```diff
@@ -690,13 +690,14 @@ def write_desc_input(filename, inputs, header=""):
             )
 
         f.write("\n# solver tolerances\n")
+        defaults = {"ftol": "1e-2", "xtol": "1e-6", "gtol": "1e-6", "nfev": "100"}
         for key in ["ftol", "xtol", "gtol", "nfev"]:
             f.write(
                 key
                 + " = {}\n".format(
                     ", ".join(
                         [
-                            str(inp[key]) if inp[key] is not None else str(0)
+                            str(inp[key]) if inp[key] is not None else defaults[key]
```

Comment:
I'm wondering if it might be better to not write these tolerances at all if they're not assigned. Everywhere else in the code we have defaults=`None` for tolerances, because the default tolerances get assigned here: https://github.com/PlasmaControl/DESC/blob/83ae6d1b0ed17d71dceb3d9528c803a75ef89c18/desc/optimize/optimizer.py#L507 and in theory we might want different defaults for different optimizers etc. When parsing an input file, these default to `None` as well if the field isn't present: https://github.com/PlasmaControl/DESC/blob/83ae6d1b0ed17d71dceb3d9528c803a75ef89c18/desc/io/input_reader.py#L170 so that would correctly propagate into the optimizer where the defaults get set. This way we only have a single place where we assign default values so it's hopefully easier to maintain.

Reply:
Yea that is probably the better way to approach this